draft-sbn-dnsop-svcb-ech.md

```diff
@@ -0,0 +1,92 @@
+
+# Client behavior {#ech-client-behavior}
+
+## Disabling fallback
+
+The SVCB-optional client behavior specified in ({{Section 3 of !SVCB}}) permits clients to fall back to a direct connection if all SVCB options fail.  This behavior is not suitable for ECH, because fallback would negate the privacy benefits of ECH.  Accordingly, ECH-capable SVCB-optional clients MUST switch to SVCB-reliant connection establishment if SVCB resolution succeeded (as defined in {{Section 3 of !SVCB}}) and all alternative endpoints have an "ech" SvcParam.
+
+## ClientHello construction
+
+When ECH is in use, the TLS ClientHello is divided into an unencrypted "outer" and an encrypted "inner" ClientHello.  The outer ClientHello is an implementation detail of ECH, and its contents are controlled by the ECHConfig in accordance with {{ECH}}.  The inner ClientHello is used for establishing a connection to the service, so its contents may be influenced by other SVCB parameters.  For example, the requirements related to ALPN protocol identifiers in {{Section 7.1.2 of SVCB}} apply only to the inner ClientHello.  Similarly, it is the inner ClientHello whose Server Name Indication identifies the desired service.
+
+## Performance optimizations
+
+Prior to retrieving the SVCB records, the client does not know whether they contain an "ech" parameter.  As a latency optimization, clients MAY prefetch DNS records that will only be used if this parameter is not present (i.e. only in SVCB-optional mode).
+
+The "ech" SvcParam alters the contents of the TLS ClientHello if it is present.  Therefore, clients that support ECH MUST NOT issue any TLS ClientHello until after SVCB resolution has completed.  (See {{Section 5.1 of !SVCB}}).
+
+# Interaction with HTTP Alt-Svc
+
+HTTP clients that implement both HTTP Alt-Svc {{?RFC7838}} and the HTTPS record type {{!SVCB}} can use them together, provided that they only perform connection attempts that are "consistent" with both sets of parameters ({{Section 9.3 of !SVCB}}).  At the time of writing, there is no defined parameter related to ECH for Alt-Svc.  Accordingly, a connection attempt that uses ECH is considered "consistent" with an Alt-Svc Field Value that does not mention ECH.
+
+Origins that publish an "ech" SvcParam in their HTTPS record SHOULD also publish an HTTPS record with the "ech" SvcParam for every alt-authority offered in its Alt-Svc Field Values.  Otherwise, clients might reveal the name of the server in an unencrypted ClientHello to an alt-authority.
+
+If all HTTPS records for an alt-authority contain "ech" SvcParams, the client MUST adopt SVCB-reliant behavior (as in {{disabling-fallback}}) for that RRSet.  This precludes the use of certain connections that Alt-Svc would otherwise allow, as discussed in {{Section 9.3 of !SVCB}}.
+
```

Comment:
It would be good to replicate some of the "Multi CDN" section from the SVCB draft, or at least the examples.  Which could just be dropping most or all of Multi-CDN {#multicdn}" back in here.  It's non-normative so repeating it might help provide clarity. There are some clarifications there that are important to ECH and which might otherwise be lost in subtlety.

Reply:
That seems fine to me, although duplicating the whole example in both drafts seems excessive.  Maybe we can shrink both copies to focus on the relevant aspects.

Perhaps we can land this PR and you can add/adjust the example before we publish -00?